fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/DruidPartialDataRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers
+
+import com.yahoo.bard.webservice.config.SystemConfig
+import com.yahoo.bard.webservice.config.SystemConfigProvider
+import com.yahoo.bard.webservice.druid.model.datasource.DataSource
+import com.yahoo.bard.webservice.druid.model.filter.Filter
+import com.yahoo.bard.webservice.druid.model.having.Having
+import com.yahoo.bard.webservice.druid.model.orderby.LimitSpec
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery
+import com.yahoo.bard.webservice.druid.model.query.Granularity
+import com.yahoo.bard.webservice.druid.model.query.GroupByQuery
+import com.yahoo.bard.webservice.druid.model.query.QueryContext
+import com.yahoo.bard.webservice.web.DataApiRequest
+import com.yahoo.bard.webservice.web.responseprocessors.DruidPartialDataResponseProcessor
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseProcessor
+
+import spock.lang.Specification
+
+class DruidPartialDataRequestHandlerSpec extends Specification {
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    def "New query context is passed to next handler"() {
+        given:
+        SystemConfig systemConfig = SystemConfigProvider.getInstance()
+        String uncoveredKey = SYSTEM_CONFIG.getPackageVariableName("druid_uncovered_interval_limit")
+        systemConfig.setProperty(uncoveredKey, '10')
+
+        DataRequestHandler nextHandler = Mock(DataRequestHandler)
+        ResponseProcessor responseProcessor = Mock(ResponseProcessor)
+        RequestContext requestContext = Mock(RequestContext)
+        DruidAggregationQuery druidQuery = Mock(DruidAggregationQuery)
+        QueryContext queryContext = Mock(QueryContext)
+        DataApiRequest apiRequest = Mock(DataApiRequest)
+
+        druidQuery.getContext() >> queryContext
+
+        DruidPartialDataRequestHandler druidPartialDataRequestHandler = new DruidPartialDataRequestHandler(
+                nextHandler
+        )
+
+        DruidAggregationQuery druidAggregationQuery = new GroupByQuery(
+                Mock(DataSource),
+                Mock(Granularity),
+                Mock(Collection),
+                Mock(Filter),
+                Mock(Having),
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Mock(Collection),
+                Mock(LimitSpec),
+                queryContext,
+                false
+        )
+
```

Comment:
When doing this I didn't realize I'd orphaned this line of code.  It's not being used.